portsyncd/portsyncd.cpp

```diff
@@ -14,50 +14,136 @@
 }
 
+void handlePortConfigFile(ProducerTable &p, string file);
+void handleVlanIntfFile(string file);
+
 int main(int argc, char **argv)
 {
     int opt;
     string port_config_file = DEFAULT_PORT_CONFIG_FILE;
+    string vlan_interfaces_file = DEFAULT_VLAN_INTERFACES_FILE;
 
-    while ((opt = getopt(argc, argv, "f:h")) != -1 )
+    while ((opt = getopt(argc, argv, "p:v:h")) != -1 )
     {
         switch (opt)
         {
-        case 'f':
+        case 'p':
             port_config_file.assign(optarg);
             break;
+        case 'v':
+            vlan_interfaces_file.assign(optarg);
+            break;
         case 'h':
-            usage(argv);
+            usage();
             return 1;
         default: /* '?' */
-            usage(argv);
+            usage();
             return EXIT_FAILURE;
         }
     }
 
     DBConnector db(0, "localhost", 6379, 0);
     ProducerTable p(&db, APP_PORT_TABLE_NAME);
 
-    ifstream infile(port_config_file);
-    if (!infile.is_open())
+    LinkSync sync(&db);
+    NetDispatcher::getInstance().registerMessageHandler(RTM_NEWLINK, &sync);
+    NetDispatcher::getInstance().registerMessageHandler(RTM_DELLINK, &sync);
+
+    try
     {
-        cerr << "Port configuration file not found! " << port_config_file << endl;
-        usage(argv);
+        NetLink netlink;
+        Select s;
+
+        netlink.registerGroup(RTNLGRP_LINK);
+        cout << "Listen to link messages..." << endl;
+        netlink.dumpRequest(RTM_GETLINK);
+
+        handlePortConfigFile(p, port_config_file);
+
+        s.addSelectable(&netlink);
+        while (true)
+        {
+            Selectable *temps;
+            int tempfd, ret;
+            ret = s.select(&temps, &tempfd, 1);
+
+            if (ret == Select::ERROR)
+            {
+                cerr << "Error had been returned in select" << endl;
+                continue;
+            }
+
+            if (ret == Select::TIMEOUT)
+            {
+                if (!g_init && g_portSet.empty())
+                {
+                    /*
+                     * After finishing reading port configuration file and
+                     * creating all host interfaces, this daemon shall send
+                     * out a signal to orchagent indicating port initialization
+                     * procedure is done and other application could start
+                     * syncing.
+                     */
+                    FieldValueTuple finish_notice("lanes", "0");
+                    vector<FieldValueTuple> attrs = { finish_notice };
+                    p.set("ConfigDone", attrs);
+
+                    handleVlanIntfFile(vlan_interfaces_file);
+
+                    g_init = true;
+                }
+            }
+        }
+    }
+    catch (...)
+    {
+        cerr << "Exception had been thrown in deamon" << endl;
         return EXIT_FAILURE;
     }
 
+    return 1;
```

Comment:
actually the program shall never go to here because there's a while(true) loop above it.